Creation of Subscribers' API (WIP)

jclarkin opened the pull request:
- Work In Progress
- Attached Create, Read (one or list), and Delete to the router
- Responses need cleanup, and does not currently comply with proper JSON API format

jclarkin committed:
Creation of Subscribers' API

* Work In Progress
* Attached Create, Read (one or list), and Delete to the router
* Responses need cleanup, and does not currently comply with proper JSON API format

jclarkin commented:
Looking for feedback on first Golang code.

I know - Needs refactoring for test-ability.

snocorp commented on src/notify/subscription/subscriptionDB.go:
```diff
@@ -0,0 +1,133 @@
+	var err error
+
+	prepStmts.insert, err = myDB.Prepare(`
+		INSERT INTO subscriptions
```
SQL in your code is fun.

asmaloney commented on src/notify/subscription/db.go:
```diff
@@ -0,0 +1,133 @@
+                id UUID PRIMARY KEY,
+                event_type VARCHAR(64),
+                context VARCHAR(64),
+                original_account_id VARCHAR(64),
```
I believe this will be a UUID.  That's what I changed events to.

asmaloney commented on src/notify/subscription/db.go:
```diff
@@ -0,0 +1,133 @@
+      original_account_id
+    )
+    VALUES ( uuid_generate_v4(), $1, $2, $3 )
+  `)
```
My formatting on these came out all wonky too...

jclarkin replied:
They look fine in my editor... just not on github.

![image](https://cloud.githubusercontent.com/assets/678856/8605406/1fc7f392-2653-11e5-84bb-3af01dc7ce13.png)

asmaloney replied:
Yeah - me too.  Don't know why.

asmaloney commented on src/notify/subscription/handlers.go:
```diff
@@ -0,0 +1,81 @@
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(201)
+	fmt.Fprintf(w, "%v", string(subscriptionJSON))
```
These lines can be replaced with `jsonUtils.Output(w, 201, subscription)`.  See the event handlers.

(Same with the others below.)

jclarkin committed:
Changed data structure from 'original_account' to 'account'

jclarkin committed:
Subscribe code simplifications based on code review
- Change of data model type to UUID
- Adoption of simplifying jsonUtils functions
